deploy.md

```diff
@@ -1,59 +1,142 @@
+## Update repository
 
-- update remote repository
+  ```
+  # merge changes to master
+  git checkout master
+  git merge release
+  git push upstream master
 
-`git push origin master`
+  # note that you might want to cherry-pick commits from the release branch if there exists some commits
+  # that are not supposed to be merged to master, for example, bugfixes for the deprecated features in the master branch.
+  git checkout master
+  git cherry-pick some_commit_hash
+  git push upstream master
 
-`git push origin --tags`
+  # push the release branch to remote
+  git push upstream release
+  git push upstream --tags
+  ```
 
 # Deploy
 
-- create / instantiate virtual env. Note the path name. it is important so that the python virtual env doesn't get included in the addon.
+We use `fabric` in a Python virtual environment to do the job,
 
-`virtualenv activity-streams-env`
+- create / instantiate virtual env. Note the path name, it is important so that the python virtual env doesn't get included in the addon
 
-`source ./activity-streams-env/bin/activate`
+  If it is the __first__ time to deploy on your machine,
+  ```
+  virtualenv activity-streams-env
+  source ./activity-streams-env/bin/activate
+  pip install -r requirements.txt
+  ```
 
-`pip install -r requirements.txt`
+  Otherwise,
+  ```
+  source ./activity-streams-env/bin/activate
+  ```
 
 - deploy
 
-Important: Ensure there are no extra files in the directory. Any file that is not in .jpmignore will be included in the addon.
-Make extra sure the .amo_config.json is named properly, otherwise it might be included in the addon XPI
+  We will deploy off the `release` branch,
+
+  ```
+  git checkout release
+  ```
+
+  Important: Ensure there are no extra files in the directory. Any file that is not in .jpmignore will be included in the addon. Make extra sure that the `.amo_config.json` is named properly, otherwise it might be included in the addon XPI.
+
+  You can make extra sure there are no extra files by running:
+  ```
+  npm run bundle  # building static assets
+  ./node_modules/.bin/jpm xpi # create addon package
+  unzip -t @activity-streams-1.X.Y.xpi | less # check for unexpected files
+  rm \@activity-streams-*.xpi # remove the addon. don't forget!
+  ```
+
+  To deploy, you need both the AWS creds and the AMO creds. If you have them, simply run:
+
+  ```
+  fab deploy
+  ```
+
+# Post-deploy
 
-You can make extra sure there are no extra files by running:
-`npm run bundle  # building static assets`
+- hotfixes
 
-`./node_modules/.bin/jpm xpi # create addon package`
+  We may find futher issues after the release is out. You can create hotfixes off the `release` branch as follows,
 
-`unzip -t @activity-streams-1.0.16.xpi | less # check for unexpected files`
+  ```
+  git checkout -b hotfix-fix-issue release
+  # fix the bug and submit PR
 
-`rm \@activity-streams-*.xpi # remove the addon. don't forget!`
+  git checkout release
+  git merge hotfix-fix-issue
```

Comment:
Yes, there is a comment for that above. Will re-organize it a bit to clarify it.